res/controllers/Hercules-P32-scripts.js

```diff
@@ -218,127 +229,164 @@ P32.Deck = function (deckNumbers, channel) {
 
     this.loopIn = new components.Button({
         midi: [0x90 + channel, 0x50],
-        inKey: 'loop_in',
+        key: 'loop_in',
+        on: P32.padColors.red,
+        off: P32.padColors.purple,
     });
     this.loopOut = new components.Button({
         midi: [0x90 + channel, 0x51],
-        inKey: 'loop_out',
+        key: 'loop_out',
+        on: P32.padColors.red,
+        off: P32.padColors.purple,
     });
-    this.loopTogglePad = new components.LoopToggleButton({
+    this.reloop = new components.Button({
         midi: [0x90 + channel, 0x52],
+        unshift: function () {
+            this.inKey = 'reloop_toggle';
+        },
+        shift: function () {
+            this.inKey = 'reloop_andstop';
+        },
         on: P32.padColors.red,
         off: P32.padColors.blue,
+        outKey: 'loop_enabled',
     });
-    this.loopIn.send(P32.padColors.purple);
-    this.loopOut.send(P32.padColors.purple);
 
     this.tempSlow = new components.Button({
         midi: [0x90 + channel, 0x44],
-        inKey: 'rate_temp_down',
-        onlyOnPress: false,
+        key: 'rate_temp_down',
+        on: P32.padColors.red,
+        off: P32.padColors.purple,
     });
     this.tempFast = new components.Button({
         midi: [0x90 + channel, 0x45],
-        inKey: 'rate_temp_down',
-        onlyOnPress: false,
+        key: 'rate_temp_up',
+        on: P32.padColors.red,
+        off: P32.padColors.purple,
     });
     this.alignBeats = new components.Button({
         midi: [0x90 + channel, 0x46],
-        inKey: 'beats_translate_curpos',
-    });
-    this.quantize = new components.Button({
-        midi: [0x90 + channel, 0x47],
-        key: 'quantize',
+        key: 'beats_translate_curpos',
         on: P32.padColors.red,
         off: P32.padColors.blue,
     });
-    this.tempSlow.send(P32.padColors.purple);
-    this.tempFast.send(P32.padColors.purple);
-    this.alignBeats.send(P32.padColors.blue);
+
+    this.enableEffectUnitButtons = new components.ComponentContainer(); //fii
+    this.enableEffectUnitButtons[1] = new components.EffectAssignmentButton({
```

Comment:
A loop would require some math for calculating the second byte of the MIDI messages, which I think is a bit overcomplicated when there are only 4 iterations. Sometimes it's easier to just copy & paste.